src/command_frame.cc

```diff
@@ -21,51 +34,84 @@ auto intelligence::jsonschema::cli::frame(
-    std::cout << "    Schema           : " << entry.root.value_or("<ANONYMOUS>")
-              << "\n";
-    std::cout << "    Pointer          :";
-    if (!entry.pointer.empty()) {
-      std::cout << " ";
-    }
+  const auto output_json = options.contains("json") || options.contains("j");
+  if (output_json) {
+    auto output_json_object = sourcemeta::jsontoolkit::JSON::make_object();
+    auto frame_json = sourcemeta::jsontoolkit::JSON::make_object();
+    auto references_json = sourcemeta::jsontoolkit::JSON::make_object();
 
-    sourcemeta::jsontoolkit::stringify(entry.pointer, std::cout);
-    std::cout << "\n";
-    std::cout << "    Base URI         : " << entry.base << "\n";
-    std::cout << "    Relative Pointer :";
-    if (!entry.relative_pointer.empty()) {
-      std::cout << " ";
+    for (const auto &[key, entry] : frame) {
+      auto frame_entry = sourcemeta::jsontoolkit::JSON::make_object();
+      frame_entry.assign("schema", sourcemeta::jsontoolkit::JSON{entry.root.value_or("<ANONYMOUS>")});
+      std::ostringstream pointer_stream;
+      sourcemeta::jsontoolkit::stringify(entry.pointer, pointer_stream);
+      frame_entry.assign("pointer", sourcemeta::jsontoolkit::JSON{pointer_stream.str()});
+      frame_entry.assign("baseURI", sourcemeta::jsontoolkit::JSON{entry.base});
+      frame_entry.assign("type", sourcemeta::jsontoolkit::JSON{enum_to_string(entry.type)});
+      std::ostringstream reference_stream;
+      sourcemeta::jsontoolkit::stringify(entry.relative_pointer, reference_stream);
+      frame_entry.assign("relativePointer", sourcemeta::jsontoolkit::JSON{reference_stream.str()});
+      frame_entry.assign("dialect", sourcemeta::jsontoolkit::JSON{entry.dialect});
+      frame_json.assign(key.second, sourcemeta::jsontoolkit::JSON{frame_entry});
     }
+    output_json_object.assign("frames", sourcemeta::jsontoolkit::JSON{frame_json});
 
-    sourcemeta::jsontoolkit::stringify(entry.relative_pointer, std::cout);
-    std::cout << "\n";
-    std::cout << "    Dialect          : " << entry.dialect << "\n";
-  }
-
-  for (const auto &[pointer, entry] : references) {
-    std::cout << "(REFERENCE) URI: ";
-    sourcemeta::jsontoolkit::stringify(pointer.second, std::cout);
-    std::cout << "\n";
-
-    std::cout << "    Type             : ";
-    if (pointer.first == sourcemeta::jsontoolkit::ReferenceType::Dynamic) {
-      std::cout << "Dynamic";
-    } else {
-      std::cout << "Static";
+    for (const auto &[pointer, entry] : references) {
+      auto ref_entry = sourcemeta::jsontoolkit::JSON::make_object();
+      ref_entry.assign("type",
+                       sourcemeta::jsontoolkit::JSON{pointer.first == sourcemeta::jsontoolkit::ReferenceType::Dynamic ? "Dynamic" : "Static"});
+      ref_entry.assign("destination", sourcemeta::jsontoolkit::JSON{entry.destination});
+      if (entry.base.has_value()) {
+        ref_entry.assign("fragmentBaseURI", sourcemeta::jsontoolkit::JSON{entry.base.value()});
```

Comment:
Here you can make it just `base`. I think in general keeping the JSON properties consistent with the names in the struct will make it easier to maintain